readfig.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Simple program feeding raw data into device
```

Comment:
Since the performance is not the key consideration, you can rewrite the program in Python.